tree.go

```diff
@@ -380,10 +385,15 @@ func (n *InternalNode) InsertStem(stem []byte, values [][]byte, resolver NodeRes
 		// splits.
 		return n.InsertStem(stem, values, resolver)
 	case *LeafNode:
-		n.cowChild(nChild)
 		if equalPaths(child.stem, stem) {
+			// We can't insert anything into a POA leaf node.
+			if child.isPOAStub {
+				return errIsPOAStub
+			}
+			n.cowChild(nChild)
```

Comment:
I don't quite like this comment, because it sounds like you can not insert in ANY type of PoA `LeafNode`, and the fact is that you can. I would rephrase as "you can't update leaf values of a PoA stem" or something like taht.

Reply:
> because it sounds like you can not insert in ANY type of PoA LeafNode, and the fact is that you can

You can't. In today's definition of `isPOAStub`, the only case is `true` is for a `LeafNode` that is only there to prove the absence of a stem. In this case, you can't insert any value.

Maybe you're thinking about potential future "stub" cases where leaf nodes are partially filled and you _might_ be able to insert things?

Reply:
yes, the way you can is precisely when you are inserting a new leaf that has a common prefix with the PoA stem. So you can not insert _values_ into it, but you can insert a `LeafNode` into it (by producing intermediate `InternalNode`s).

Reply:
Right, but my comment is inside `equalPaths(child.stem, stem)` so I'm not talking about that case, but inserting values.

Reply:
In any case, I can add `insert values` in the comment if that helps.

Reply:
Done.